### ISSUE TITLE
Creating a Prompts.choice fails with 500 Internal Server Error

### ISSUE THREAD

**Issue by JamieREvans** [Microsoft/BotBuilder#891]
I've written code based on [the prompts documentation](https://docs.botframework.com/en-us/node/builder/chat/prompts/#promptschoice) for offering suggested responses to users, but it's crashing every time I send the suggested responses.

Here is my code:

``` javascript
var express = require('express');
var builder = require('botbuilder');

var router = express.Router();
var app = express();

var connector = new builder.ChatConnector({
    appId: 'app-id',
    appPassword: 'password'
});

router.route('/api/messages')
  .post(connector.listen());

app.use('/', router);

var bot = new builder.UniversalBot(connector);

bot.dialog('/', [
  function(session) {
    builder.Prompts.choice(session, 'Are you a boy or girl?', ['boy', 'girl']);
  },
  function(session, results) {
    session.send('Thanks!');
  }]);

module.exports = app;
```

and here is the error stacktrace

```
Error: Request to 'https://kik.botframework.com/v3/conversations/f9370f706cee0f9b385fb96acbd4c56d4a8ddbaecc76300e64b8b357f8d58ad3/activities/3JgiVY2dkr6' failed: [500] Internal Server Error
    at Request._callback (/~path~/microsoft-chatbot/node_modules/botbuilder/lib/bots/ChatConnector.js:356:46)
    at Request.self.callback (/~path~/microsoft-chatbot/node_modules/request/request.js:187:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/~path~/microsoft-chatbot/node_modules/request/request.js:1044:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/~path~/microsoft-chatbot/node_modules/request/request.js:965:12)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
```

I'm using Kik to send the messages to my bot. I'm wondering if it may be a bug with Microsoft's Kik adapter for sending suggested responses.

**Comment by gjrevans** [comment]
👍

**Comment by Stevenic** [comment]
Adding Lars... Any thoughts? Maybe you don't like the Card I'm sending you?

**Comment by LarsLiden** [comment]
Can you share the JSON payload generated by the code?

**Comment by JamieREvans** [comment]
```json
   {  
      "type":"message",
      "text":"Are you a boy or girl?",
      "attachments":[  
         {  
            "contentType":"application/vnd.microsoft.card.hero",
            "content":{  
               "buttons":[  
                  {  
                     "type":"imBack",
                     "value":"boy",
                     "title":"girl"
                  },
                  {  
                     "type":"imBack",
                     "value":"girl",
                     "title":"girl"
                  }
               ]
            }
         }
      ],
      "from":{  
         "id":"kikchatbotname",
         "name":"botname"
      },
      "recipient":{  
         "id":"mykikid",
         "name":"mykikname"
      }
   }
```

**Comment by LarsLiden** [comment]
Kik expects the hero card to actually contain text and will reject cards that have null or empty text fields.  @Stevenic , rather than setting the "text" field in the outer message, can you set the "text" field in the herocard?

**Comment by LarsLiden** [comment]
This bug has been resolved in the Kik channel service and the fix will be published in our next push to production.

**Comment by JamieREvans** [comment]
@LarsLiden @Stevenic This is still happening randomly with Kik only.

I am sending an identical payload each time, but I will sometimes get a `Error: Request to 'https://kik.botframework.com/v3/conversations/cd83e74eab7fcd54b72ce84e06bbaeea9991178e52514bf75d1eaa66c7982d4a/activities/1zcMjEEX11u' failed: [500] Internal Server Error`. It seems to happen about 5-10% of the time.

I've never had this happen with any other connector.

**Comment by Stevenic** [comment]
We'll try to look into the issue.

**Comment by yuyuvn** [comment]
facebook also have this issue, especially when send a big image
```
Error: Request to 'https://facebook.botframework.com/v3/conversations/914073512035535-277316602635874/activities/mid.1472014370271%3A87a90161cb2402c453' failed: [500] Internal Server Error
```

**Comment by JamieREvans** [comment]
I'm getting the same issue with Facebook as @yuyuvn. Seems all adapters are occasionally out of commission?

Is there any update?

**Comment by enby** [comment]
I get the same issue that @yuyuvn does, where large images (or images on slow servers) cause a generic 500 error almost every time. Here's my theory of what's happening:
- Connector sends a formatted request to the respective endpoint (for example kik api)
- Kik's servers fail to load the image within a reasonable amount of time and either don't respond to the connector in time, or respond with an error state that the connector doesn't understand

Note that this is 100% reproducible when specifying a URL that doesn't exist, and 80% for cases where image is either large or slow to load.

My assumption is that this also applies to Prompts - they simply timeout (maybe request timeouts are way too aggressive on the connector end?), and the end developer receives a generic 500 code

**Comment by LarsLiden** [comment]
Looks like we had some code that was masking bad data being passed to Kik and instead returning a 500. Made a change so the error (i.e. invalid URL) gets properly returned.   Still investigating large image problem.

**Comment by LarsLiden** [comment]
I haven't been able to reproduce the large image issue.  How big of an image are you using?  Can you point me to one?

**Comment by dandriscoll** [comment]
Hi all, it looks like we've resolved the first issue and are waiting on additional details for the second.

@enby, if you're able to reproduce the image problem and can point us to an example that fails, please open a new issue. Thanks much
